core/src/main/java/org/elasticsearch/node/Node.java

```diff
@@ -188,53 +197,82 @@
      * @param preparedSettings Base settings to configure the node with
      */
     public Node(Settings preparedSettings) {
-        this(InternalSettingsPreparer.prepareEnvironment(preparedSettings, null), Collections.<Class<? extends Plugin>>emptyList());
+        this(InternalSettingsPreparer.prepareEnvironment(preparedSettings, null));
     }
 
-    protected Node(Environment tmpEnv, Collection<Class<? extends Plugin>> classpathPlugins) {
-        Settings tmpSettings = Settings.builder().put(tmpEnv.settings())
-                .put(Client.CLIENT_TYPE_SETTING_S.getKey(), CLIENT_TYPE).build();
+    public Node(Environment environment) {
+        this(environment, Collections.emptyList());
+    }
+
+    protected Node(final Environment environment, Collection<Class<? extends Plugin>> classpathPlugins) {
         final List<Closeable> resourcesToClose = new ArrayList<>(); // register everything we need to release in the case of an error
+        boolean success = false;
+        {
+            // use temp logger just to say we are starting. we can't use it later on because the node name might not be set
+            ESLogger logger = Loggers.getLogger(Node.class, NODE_NAME_SETTING.get(environment.settings()));
+            logger.info("initializing ...");
```

Comment:
I do not like the `_unset_` that comes out of this log message. Now there are three prefixes: empty prefix (e.g., from the jar hell check log messages when debug logging is enabled, which occur before the node is constructed, and then also occur for module and plugin loading), the node name prefix, and this `_unset_` that appears on a single line. We've accepted the empty prefix in the past, and I think we should just do so here?